docs/src/architecture/domain-model.md

```diff
@@ -0,0 +1,214 @@
+  * also the **Domain Application** can expose data as input or output (e.g. own _signals_ or _states_)
+
+## IoT Cloud Domain
+
+Cloud end-points could be IoT Cloud Platforms, as e.g. [Cumulocity](https://www.softwareag.cloud/site/product/cumulocity-iot.html), [Azure](https://azure.microsoft.com) or [AWS](https://aws.amazon.com); or end-points for specific IoT tasks, as e.g. a time-series database to collect measurements.
+
+* Each **device** is represented in the IoT cloud by an individual **device twin**
+* All information / data or functionality the device provides to the cloud appears in context of it's **device twin**
+* The focus of **thin-edge** is on two IoT Cloud aspects:
+  - 1) **Telemetry Data Handling**
+  - 2) **Device Management**
+
+### Telemetry Data Handling
+
+**Telemetry data** are the measurements, events and alarms collected or raised by sensors, actuators or domain applications.
+
+  * **Measurement**, is a single value or set of values
+    * could be be a mix of values produced by one or more sensors and values calculated by the device's domain application
+    * values could be a mix of numbers, strings or booleans
+    * has one timestamp
+    * releates to one **Metric**
+  * **Metric**, is a time-series of measurements
+    * contains a _source device_
+    * contains a _type name_
+    * optionally contains _units_ for the **measurements**
+  * TODO: Example for metric and it's measurements
+  * **Command**, is a single value or set of values
+    * is send from the cloud to one device, e.g. to
+      - stimulate an actuator (e.g. switching a relay)
+      - send a signal to the domain application
+      - set one or more set-points (e.g. upper/lower limits or threshold of a climate control)
+    * values could be a mix of numbers, strings or booleans
+  * **Event**, is a notification that something happened on the device's environment or software system
+    * it's source could be
+      - a sensor that detected a door has beed closed
+      - a signal from the device's domain application
+      - a device's system notification that a user has started an ssh session
+    * has one timestamp given by the producer, or implicitly set as the current time of receiving at **thin-edge**
+  * **Alarm**, is similar to an event, but the _End User_ (an operator of the system) has to take action to resolve the alarm.
+
+### Device Management
+
+**Device Management** provides to manage and monitor devices in the field from the cloud. That includes:
+  * **Software Management**, provides to manage the installed software packages on the device
+    * view list and versions of installed packages
+    * install new or update existing software packages
+    * remove installed software packages
+  * **Configuration Management**, provides to view and change configurations on the device
+    * lists available configuration files
+    * retrieve content of individual configuration files
```

Comment:
note that "retrieve" and "send" are terms from cloud viewpoint.
Better:?
"transfer individual configuration file from device to cloud"
(similar line below)

Reply:
Wording changed , see https://github.com/thin-edge/thin-edge.io/commit/745fe78140d561cc73e4d5420e6b72e57de31771